### ISSUE TITLE
WebsocketException reason is not visible to the client

### ISSUE THREAD

**Issue by PawelRoman** [tiangolo/fastapi#5922]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi import WebSocketException, status

app = FastAPI()

@app.websocket("/ws")
async def websocket_endpoint():
   raise WebSocketException(code=status.WS_1008_POLICY_VIOLATION, reason="I want you to know this")
```

### Description

The reason code and text seem to be unavailable for the client. The only way to get them is to have websocket client written in python, import starlette's WebSocketException and catch the exception manually. 

When making calls from other clients, e.g. from Postman, the code and reason are just not available. It would be nice if the server returned code and reason as handshake response headers e.g. X-Code and X-Reason or something. Not sure if that makes sense?

### Wanted Solution

Have code and reason from WebSocketException be surfaced as HTTP response headers

### Wanted Code

```python
Put any non-python websocket client code here, connect to websocket and try to get code and reason on failed connection.
```

### Operating System

Linux, Windows, macOS

### FastAPI Version

0.88.0

### Python Version

3.11

**Comment by Kludex** [comment]
No, it doesn't make sense. The handshake already happened when you raise the exception.

Have you tried another WebSocket client?
